test/framework/ingress.go

```diff
@@ -23,238 +24,165 @@ func (s HostAndPort) Port() string {
 	return port
 }
 
+type ingressKustomization struct {
+	ServiceName string
+	Host        string
+}
+
+type cqlIngressKustomization struct {
+	ServiceNamespace string
+	ServiceName      string
+	Host             string
+}
+
 func (f *E2eFramework) DeployStargateIngresses(t *testing.T, k8sContext, namespace, stargateServiceName string, stargateRestHostAndPort HostAndPort) {
-	src := filepath.Join("..", "..", "test", "testdata", "ingress", "stargate-ingress.yaml")
-	dest := filepath.Join("..", "..", "build", "test-config", "ingress", "stargate", k8sContext)
-	_, err := utils.CopyFileToDir(src, dest)
-	require.NoError(t, err)
-	err = generateStargateIngressKustomization(k8sContext, namespace, stargateServiceName, stargateRestHostAndPort.Host())
-	require.NoError(t, err)
-	err = f.kustomizeAndApply(dest, namespace, k8sContext)
-	require.NoError(t, err)
+	f.deployIngress(t, k8sContext, namespace, "stargate-ingress.yaml", "stargate", stargateTemplate,
+		&ingressKustomization{stargateServiceName, stargateRestHostAndPort.Host()})
+
+	f.deployIngress(t, k8sContext, "ingress-nginx", "stargate-cql-ingress.yaml", "stargate-cql", stargateCqlTemplate,
+		&cqlIngressKustomization{namespace, stargateServiceName, stargateRestHostAndPort.Host()})
```

Comment:
I used the opportunity to eliminate a bit of duplication here, all the function do essentially the same thing.